### PR TITLE
Remove licensor component

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Remove deprecated component related to self-hosted

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/16981

## Release Notes
```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [x] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/admin/License.tsx
Comment (AlexTugarev) [inline components/dashboard/src/admin/License.tsx:36]: ❓ do we need the License page at all? 
cc. @geropl and @atduarte

We can do that as a follow-up.

File: components/server/ee/src/workspace/gitpod-server-impl.ts
Comment (AlexTugarev) [inline components/server/ee/src/workspace/gitpod-server-impl.ts:354]: 👍🏻  not removing the method should work fine during roll out.

File: yarn.lock
Comment (AlexTugarev) [inline yarn.lock:12525]: 🕺🏻

**Review by AlexTugarev — APPROVED** [review]
LGTM

/hold just in case if this needs a second reviewer

**Comment by AlexTugarev** [comment]
❓ What should happen with the license handling in `./gitpod/.werft`?

**Comment by geropl** [comment]
@AlexTugarev I have no time to look into this PR, but: We still rely on there being a license/empty license for the "Dedicated" setup atm; this is what helps determine the BillingMode for Dedicated (just like self-hosted) to be `none`.

Could you check that with this PR this is still the case? :pray:

**Comment by geropl** [comment]
We just double-checked: There is no usage of `config.license` anymore. :heavy_check_mark:

**Comment by geropl** [comment]
/unhold

Deployment is :green_circle:

**Pull request merged**
